ornlneutronimaging/mcpevent2hist
# Restructure build system to use sub-packages

# Description of Pull Request

This PR adjust the CMake config such that each target now has its own cmake config file, which allows downstream application build selected target instead of everything all at once.

## Purpose of work

This is an improvement of the build system.



## Summary of work

CMake build system is improved.

## Additional detail of work


## Testing instructions

Just clone the feature branch and following the same build instruction as before.
The change should be transparent to regular users.





> IBM item: [S1594](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=1594)
